waspc/data/Generator/templates/react-app/src/router/linkHelpers.ts

```diff
@@ -0,0 +1,49 @@
+export type ParamValue = string | number
+export type Params = { [name: string]: ParamValue }
+export type Search =
+  | string[][]
+  | Record<string, string>
```

Comment:
Same story like here: https://github.com/wasp-lang/wasp/pull/1235#discussion_r1300022520.

The semantic implications of `Record`s and index signatures are a nitpick, but I think naming a key would be beneficial here.

The same applies to other similar `Record`s.

Reply:
Commented IRL: we copied the type from the built-in `URLSearchParams` class